sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -581,7 +581,9 @@ object FunctionRegistry {
         val params = Seq.fill(expressions.size)(classOf[Expression])
         val f = constructors.find(_.getParameterTypes.toSeq == params).getOrElse {
           val validParametersCount = constructors
-            .filter(_.getParameterTypes.forall(_ == classOf[Expression]))
+            .filter(_.getParameterTypes.forall { t =>
+              t == classOf[Expression] || t == classOf[Seq[Expression]]
```

Comment:
I think generic type can't be checked runtime. Also, this code path is only for `Expression`. See `f.newInstance(expressions : _*)`.

Reply:
Second that.